src/Controller/AppController.php

```diff
@@ -63,10 +63,18 @@ public function beforeRender(Event $event)
         // Note: These defaults are just to get started quickly with development
         // and should not be used in production. You should instead set "_serialize"
         // in each action as required.
-        if (!array_key_exists('_serialize', $this->viewVars) &&
-            in_array($this->response->type(), ['application/json', 'application/xml'])
-        ) {
-            $this->set('_serialize', true);
+        if (!array_key_exists('_serialize', $this->viewVars)) {
+            $mimeType = $this->RequestHandler->mapAlias(
```

Comment:
I'm not sure we can rely on `RequestHandler` being loaded. In addition to this approach you can also use `$this->response->getHeaderLine('Content-Type')` to read the content type header. That might result in simpler code that lets you continue to use `in_array()`.

Reply:
Why couldn't we rely on `RequestHandler` here ?

--> I didn't know about that one, I may simplify this if you're OK with the concept !

Reply:
https://gist.github.com/inoas/e58900d3e266a8827b988142af490c6f
... or about the same in class `\Cake\Utility\Text` may help